ktlint/build.gradle.kts

```diff
@@ -48,9 +48,13 @@ val shadowJarExecutable by tasks.registering(DefaultTask::class) {
     description = "Creates self-executable file, that runs generated shadow jar"
     group = "Distribution"
 
+    val disableSigning = (project.findProperty("ktlint.disable-signing") as? String).toBoolean()
```

Comment:
```suggestion
    val disableSigning = project.findProperty("ktlint.disable-signing") == "true"
```

Reply:
I believe you just need to amend this flag to 
https://github.com/pinterest/ktlint/blob/ec953e68c788d9fd68a79176c864e66e7bd5303a/build-logic/src/main/kotlin/ktlint-publication.gradle.kts#L101

Reply:
Test this change with:
1. Add `ktlint.disable-signing=true` in your gradle.properties.
2. Update `VERSION_NAME` to `0.49.0`.
3. Run `./gradlew publishToMavenLocal`.

Reply:
I tested for both a snapshot and a non-snapshot version:

* `./gradlew :ktlint:shadowJarExecutable` (default behavior)
* `./gradlew :ktlint:shadowJarExecutable -Pktlint.disable-signing=false`  (explicitly enabled)
* `./gradlew :ktlint:shadowJarExecutable -Pktlint.disable-signing=true`  (explicitly disabled)

Reply:
I meant you just need to update the `isRequired`.

Reply:
ah that's what you meant. makes sense.